Zilliqa/contracts/xsgd_contract.scilla

```diff
@@ -389,29 +415,17 @@ transition DecreaseMinterAllowance(minter : ByStr20, amount : Uint128, initiator
   
   (* Retrieves initial allowance of minter, calculates new total after subtracting amount from it *)
   option_mint_allowance <- minterAllowed[minter];
-  mint_allowance = get_minter_allowance option_mint_allowance;
-  new_option_mint_allowance =
-    
-    (* If amount to be subtracted > minter allowance, set minter allowance to None {Uint128} *)
-    let amount_le_allowance = uint128_le amount mint_allowance in
-    match amount_le_allowance with
-    | True =>
-        let new_mint_allowance = builtin sub mint_allowance amount in
-        let is_zero = builtin eq new_mint_allowance zero in
-        match is_zero with
-        | True => None {Uint128}
-        | False => Some {Uint128} new_mint_allowance
-        end
-    | False => None {Uint128}
-    end;
-    
+  mint_allowance = deconstruct_option_option_uint128 option_mint_allowance;
+  new_mint_allowance = better_subtract mint_allowance amount;
+  new_option_mint_allowance = construct_option_uint128 new_mint_allowance;
+
   (* Update state and emits event *)
   minterAllowed[minter] := new_option_mint_allowance;
   e = { _eventname : "DecreasedMinterAllowance"; minter : minter; new_option_mint_allowance : new_option_mint_allowance; sender : initiator};
   event e
 end
 
-transition Mint(to: ByStr20, value : Uint128, initiator : ByStr20, o_bal : Uint128, current_supply : Uint128)
+transition Mint(to: ByStr20, value : Uint128, initiator : ByStr20, to_bal : Uint128, current_supply : Uint128)
```

Comment:
Was just about to comment on `o_bal` from previous commit, seems like the bug got squashed already

Reply:
yep! This was giving me a compilation error last time but i fixed it